diff --git a/flickr.inc b/flickr.inc
index 39dcd2e..1a2e395 100644
--- a/flickr.inc
+++ b/flickr.inc
@@ -1989,7 +1989,7 @@ function flickr_curl_http_request($url, array $options = array()) {
       elseif ($options['max_redirects']) {
         // Redirect to the new location.
         $options['max_redirects']--;
-        $result = curl_http_request($location, $options);
+        $result = flickr_curl_http_request($location, $options);
         $result->redirect_code = $code;
       }
       if (!isset($result->redirect_url)) {
